hal/inc/Keypad.h

```diff
@@ -0,0 +1,43 @@
+#ifndef _KEYPAD_H_
+#define _KEYPAD_H_
+
+#define KEYPAD_DEBOUNCING_TIME 20
+#define COL_SIZE               4
+#define ROW_SIZE               4
+
+namespace avr {
+namespace hal {
+namespace keypad  {
+
+class Keypad : public avr::hal::gpio::InputDeviceInterface{
+public:
+
+    enum KeypadState { KPressed, KNotPressed };
+    using Port = avr::mcal::gpio::Gpio::Port;
+    using Pin = avr::mcal::gpio::Gpio::Pin;
+    //using DS = Gpio::DirectionState;
+    Keypad(Port port,
+           Pin c1, Pin c2, Pin c3, Pin c4,
+           Pin r1, Pin r2, Pin r3, Pin r4);
+    uint8_t KeypadGetPressed();
```

Comment:
No need to prefix method with `Keypad` just call it `GetKey()` or `GetPressedKey()`